### ISSUE TITLE
Assignment 3: Ready for review

### ISSUE THREAD

**Issue by sushmatn** [sushmatn/TwitterClient#1]
My twitter client is ready for review. /cc @codepathreview @codepath

**Comment by codepathreview** [comment]
@sushmatn Gif is missing from readme.md. Please add the same for review.

**Comment by codepathreview** [comment]
Excellent work again Sushma. A few notes after checking out the code:
- Good to see you completed all required and optionals stories. 
- Good job polishing UI
- Nice job organizing the code 
- Great job implementing the compose tweet feature as a dialog
- Nice touch with the character count being displayed while the user composes a tweet
- Consider putting a limit on number of characters allowed in edit text (android:maxLength="140")
- Good job adding new tweet automatically to timeline.
- Good job using parceable for serialization of models
- Good to see you added ActiveAndroid annotations to the models for offline usage
- Good job with the animations for favorite etc

Here's a detailed [Project 3 Feedback Guide here](http://courses.codepath.com/snippets/intro_to_android/project_3_feedback) which covers the most common issues with this submitted project. Read through the feedback guide point-by-point to determine how you could improve your submission. 

Let us know if you have any other thoughts or questions about this assignment. Hopefully by now you feel pretty comfortable with all the major pieces to basic Android apps (Views, Controllers, ActionBar, Navigation, Models, Authentication, API Communication, Persistence, et al) and see how they all fit together. We are close now to a turning point in the course where you should be hitting a "critical mass" towards your knowledge of Android.

**Comment by sushmatn** [comment]
@codepathreview My twitter client contains a progress bar in the action bar, but for some reason it does not show up. Even when I try to set it to 'visible' and never hide it, I just see the other items in the actionbar move to make room for the progress bar, but I still can't see the progress bar.  I was not able to figure this one out.

**Comment by nesquena** [comment]
> My twitter client contains a progress bar in the action bar, but for some reason it does not show up. Even when I try to set it to 'visible' and never hide it, I just see the other items in the actionbar move to make room for the progress bar, but I still can't see the progress bar. I was not able to figure this one out.

Looks like you [followed the guide correctly](https://guides.codepath.com/android/Handling-ProgressBars#progress-within-actionbar) as far as I can spot.  I wonder if something needs to be updated in the guide to make this work again. Will have to take a look, if you end up figuring this out, please report back too.

**Comment by sushmatn** [comment]
@codepathreview In the guide I noticed that we call getActionView and get the progress bar reference, but we never use it. I tried to change the visibility of the ProgressBar returned by getActionView, but that did not work either.

 @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        // Store instance of the menu item containing progress
        miActionProgressItem = menu.findItem(R.id.miActionProgress);
        // Extract the action-view from the menu item
        ProgressBar v =  (ProgressBar) MenuItemCompat.getActionView(miActionProgressItem);
        // Return to finish
        return super.onPrepareOptionsMenu(menu);
    }

**Comment by nesquena** [comment]
A colleague of mine downloaded your repo and he said "The guide works fine.  I'm also able to see it briefly on the emulator with API 19 Samsung 4.4.4 using her repo. "

![image](https://cloud.githubusercontent.com/assets/6511/10297428/8a738868-6b87-11e5-8731-a4b83ec58c67.png)

Looks like it may be working at least on API 19. What API were you running?

**Comment by sushmatn** [comment]
I'm running API 21 on both my emulator and phone.
I actually had the progress bar working on the ImageSearch app on the same
emulator and phone.

**Comment by sushmatn** [comment]
I found the problem. I changed the actionbar background to red and saw a
blue progressbar. It was blue on blue and so wasn't showing.

Then I found out that the progressbar uses the 'colorAccent'.  I had
defined the same color for my primary and accent colors.
Changing the accent color fixed the problem.

![progress_red](https://cloud.githubusercontent.com/assets/13898009/10315485/7265c244-6c0f-11e5-96f9-7d0046cebd06.png)
